src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/DefaultEditorTemplates.cs

```diff
@@ -378,6 +378,54 @@ public static IHtmlContent MonthInputTemplate(IHtmlHelper htmlHelper)
             return GenerateTextBox(htmlHelper, inputType: "month");
         }
 
+        public static IHtmlContent WeekInputTemplate(IHtmlHelper htmlHelper)
+        {
+            // A new HTML5 input type that only will be rendered in Rfc3339 mode    
+            var value = htmlHelper.ViewData.Model;
+            var metadata = htmlHelper.ViewData.ModelMetadata;
+
+            if (htmlHelper.ViewData.TemplateInfo.FormattedModelValue == value || !metadata.HasNonDefaultEditFormat)
+            {
+                if (value is DateTime || value is DateTimeOffset)
+                {
+                    var date = DateTimeOffset.Parse(Convert.ToString(value)).DateTime;
+                    var calendar = System.Threading.Thread.CurrentThread.CurrentCulture.Calendar;
+
+                    var year = calendar.GetYear(date);
+                    var month = calendar.GetMonth(date);
+                    var day = calendar.GetDayOfWeek(date);
+
+                    // Get ISO 8601 Week
+                    if (day >= DayOfWeek.Monday && day <= DayOfWeek.Wednesday)
+                    {
+                        date = date.AddDays(3);
+                    }
+
+                    var weekRule = CalendarWeekRule.FirstFourDayWeek;
+                    var firstWeekDay = DayOfWeek.Monday;
+                    var week = calendar.GetWeekOfYear(date, weekRule, firstWeekDay);
+
+                    // Last week (either 52 or 53) includes January dates (1st, 2nd, 3rd) 
+                    if (week >= 52 && month == 1)
+                    {
+                        year = year - 1;
+                    }
+
+                    // First week includes December dates (29th, 30th, 31st)
+                    if (week == 1 && month == 12)
+                    {
+                        year = year + 1;
+                    }
+
+                    htmlHelper.Html5DateRenderingMode = Html5DateRenderingMode.Rfc3339;
```

Comment:
If this is necessary, something is very busted. What requires it? (`Html5DateRenderingMode` should always be irrelevant when `FormattedModelValue` is not equal to the `Model`.)